luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/LuthierIndustrialPark.java

```diff
@@ -67,10 +72,35 @@ Dimension getDimension(String dimensionName) {
         return dimensionDictionary.findByApiName(dimensionName);
     }
 
-/*
-    SearchProvider getSearchProvider(String searchProviderName);
-    KeyValueStore getKeyValueStore(String keyValueStoreName);
-*/
+    /**
+     * Bare minimum that can work
+     */
+
+    // TODO: Magic values!
+    int MAGIC_queryWeightLimit = 10000;
+    String MAGIC_luceneIndexPath = "path";
+    int MAGIC_maxResults = 10000;
+    public SearchProvider getSearchProvider(String searchProviderName) {
+        switch (searchProviderName) {
+            case "com.yahoo.bard.webservice.data.dimension.impl.NoOpSearchProvider":
+                return new NoOpSearchProvider(MAGIC_queryWeightLimit);
+            case "com.yahoo.bard.webservice.data.dimension.impl.LuceneSearchProvider":
+                return new LuceneSearchProvider(MAGIC_luceneIndexPath, MAGIC_maxResults);
+            default:
+                return new ScanSearchProvider();
+        }
+    }
+
+    public KeyValueStore getKeyValueStore(String keyValueStoreName) throws UnsupportedOperationException {
+        switch (keyValueStoreName) {
+            // TODO: Magic values!
+            case "com.yahoo.bard.webservice.data.dimension.RedisStore":
+                throw new UnsupportedOperationException(keyValueStoreName);
+            default:
+                return new MapStore();
+        }
```

Comment:
When you build the KeyValueStore.lua, these constructor calls should be wrapped by a KeyValueStoreFactory.
The map of named KVS Factories should be passed into the LuthierIndustrialPark via constructor.

Reply:
also, there should be no default factory.

Reply:
Please see here:
https://github.com/yahoo/fili/pull/919#discussion_r293970424